### PR TITLE
feat: Argument spec implementation for selinux role

### PR DESCRIPTION
**Pull request by DonatSzabo**
Enhancement: Added argument spec and assert role spec validation to the selinux role. Also wrote tests for it found in tests/tests_invalid_input.

Reason: Because it is a good addition to the linux-system-roles project.

Result: Successfully added it and prepared tests for it. I used AI during this implementation.

Issue Tracker Tickets (Jira or BZ if any): https://github.com/linux-system-roles/postfix/issues/206 https://redhat.atlassian.net/browse/RHELMISC-16008

Notes: Added selinux_modules to defaults/main.yml because it was missing.

## Summary by CodeRabbit

- **New Features**
  - Added configuration support for managing SELinux policy modules, including installation, enabling, disabling, and removal.
  - Added comprehensive documentation for SELinux role options, defaults, accepted values, and nested configuration structures.
  - File context types and login users can now be omitted when not required.

- **Bug Fixes**
  - Added validation that detects invalid SELinux settings and incomplete file context, login, and policy module definitions before execution.

- **Tests**
  - Added coverage for invalid inputs and supported edge cases across Ansible versions.

### CODE REVIEW

**Comment by DonatSzabo** [comment]
@coderabbitai review

**Comment by DonatSzabo** [comment]
[citest_all]

File: meta/argument_specs.yml [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+          state:
+            type: str
+            choices:
+              - enabled
+              - disabled
+              - absent
+            default: enabled
+            description: >
+              Whether the module should be installed or enabled
+              (`enabled`), disabled (`disabled`), or removed (`absent`).
```
Comment (richm) [inline meta/argument_specs.yml]: You will need to rebase on top of the latest `main` branch to pick up the new selinux_modules option `ignore_module_cache`
